### ISSUE TITLE
target-counter doesn't seem to work correctly

### ISSUE THREAD

**Issue by yurtsevich** [Kozea/WeasyPrint#1847]
WeasyPrint v58.1 is used via command line.

CSS for page numbers in footer:
`.footer.page-number .number::after {
    counter-increment: page;
    content: counter(page) " |";
    margin-right: 5px;
    color: black;
}`

And it's displayed correctly:

![2023-03-23_15-16](https://user-images.githubusercontent.com/485579/227234665-9c57fe84-da93-45d6-900f-cb19f4a16aeb.png)

CSS for page numbers in TOC:
`ul.toc li > a.page-number::after {
    content: attr(href) " - " target-counter(attr(href), page);
}`

It doesn't seem to work, giving '1' as page number for absolutely all elements in TOC:

![2023-03-23_15-15](https://user-images.githubusercontent.com/485579/227235170-92200c20-7004-4cd5-83fe-c58f4b8b9ff3.png)

Please notice that anchor used in `target-counter` is correct, referencing right element in document, also jumping to it when page number clicked.

**Comment by liZe** [comment]
Hi!

Could you please send a sample reproducing your problem? I don’t know how your footer is repeated, and I can’t reproduce this behavior with a `position: fixed` property. You can send a mail if you don’t want to share your documents publicly.

**Comment by yurtsevich** [comment]
> Hi!
> 
> Could you please send a sample reproducing your problem? I don’t know how your footer is repeated, and I can’t reproduce this behavior with a `position: fixed` property. You can send a mail if you don’t want to share your documents publicly.

Hi, thank you for the response.

I wanted to use target-counter approach which was mentioned here: https://github.com/Kozea/WeasyPrint/pull/652#issuecomment-403276559

I've compared code from example above and my code and found out myself that the problem was that I used counter-reset + counter-increment as usually it's done when using counters in CSS.

So, the issue is sorted. Sorry for disturbing
